spyder/plugins/updatemanager/workers.py

```diff
@@ -118,28 +187,54 @@ def __init__(self, stable_only):
         super().__init__()
         self.stable_only = stable_only
         self.latest_release = None
-        self.releases = None
         self.update_available = False
         self.error = None
+        self.channel = None
 
-    def _check_update_available(self):
+    def _check_update_available(self, releases, github=True):
         """Checks if there is an update available from releases."""
-        # Filter releases
-        releases = self.releases.copy()
         if self.stable_only:
             # Only use stable releases
-            releases = [r for r in releases if is_stable_version(r)]
-        logger.debug(f"Available versions: {self.releases}")
-
-        self.latest_release = releases[-1] if releases else __version__
-        self.update_available = check_version(
-            __version__,
-            self.latest_release,
-            '<'
-        )
+            releases = [r for r in releases if not r.is_prerelease]
+        logger.debug(f"Available versions: {releases}")
+
+        latest_release = max(releases) if releases else CURRENT_VERSION
+        update_available = CURRENT_VERSION < latest_release
+
+        logger.debug(f"Latest release: {latest_release}")
+        logger.debug(f"Update available: {update_available}")
+
+        # Check if the asset is available for download.
+        # If the asset is not available, then check the next latest
+        # release, and so on until either a new asset is available or there
+        # is no update available.
+        if github:
+            asset_available = False
+            while update_available and not asset_available:
+                asset_info = get_asset_info(latest_release)
+                page = requests.head(asset_info['url'])
+                if page.status_code == 302:
+                    # The asset is found
+                    logger.debug(f"Asset available for url: {page.url}")
+                    asset_available = True
+                else:
+                    # The asset is not available
+                    logger.debug(
+                        "Asset not available: "
+                        f"{page.status_code} Client Error: {page.reason}"
+                        f" for url: {page.url}"
```

Comment:
```suggestion
                        f"{page.status_code} Client Error: {page.reason} "
                        f"for url: {page.url}"
```

Strings that are composed of several lines are easier to read when the space is left at end of a line, not at the beginning of the next.